app/controllers/DetailedIncomeTaxEstimateController.scala

```diff
@@ -61,7 +56,7 @@ class DetailedIncomeTaxEstimateController @Inject()(
           Right(taxCodeIncomes),
           taxAccountSummary,
           codingComponents,
-          TaiSuccessResponseWithPayload(nonTaxCodeIncome: NonTaxCodeIncome)
+          nonTaxCodeIncome
```

Comment:
We have a handler for any exceptions here, will this be enough?

Reply:
I don't think it's necessary as if it doesn't pattern match it will default to a MatchException which is a NonFatal one

Reply:
if `nonTaxCodeIncomes` fails it will be recovered in the `case NonFatal` so it should be fine